packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -3934,6 +3953,7 @@ namespace Ifpack2 {
         } else {
           sweep_step_ = sweep_step_upper_bound_;
         }
+        IFPACK2_BLOCKTRIDICONTAINER_TIMER_FENCE(typename ImplType<MatrixType>::execution_space)
```

Comment:
Looks like everything is serial here, do you worry that other things prior to this function call might be non-blocking? In that case I would suggest we control the flow in that location instead of here.